plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -104,27 +102,46 @@ func init() {
 	inputs.Add("snmp_trap", func() telegraf.Input {
 		return &SnmpTrap{
 			timeFunc:       time.Now,
+			lookupFunc:     lookup,
 			ServiceAddress: "udp://:162",
 			Timeout:        defaultTimeout,
 			Version:        "2c",
 		}
 	})
 }
 
-func realExecCmd(timeout config.Duration, arg0 string, args ...string) ([]byte, error) {
-	cmd := exec.Command(arg0, args...)
-	var out bytes.Buffer
-	cmd.Stdout = &out
-	err := internal.RunTimeout(cmd, time.Duration(timeout))
-	if err != nil {
-		return nil, err
-	}
-	return out.Bytes(), nil
+func (s *SnmpTrap) Init() error {
+	// must init, append path for each directory, laod module for every file
+	// or gosmi will fail without saying why
+	gosmi.Init()
+	s.getMibsPath()
+	return nil
 }
 
-func (s *SnmpTrap) Init() error {
-	s.cache = map[string]mibEntry{}
-	s.execCmd = realExecCmd
+func (s *SnmpTrap) getMibsPath() error {
+	var folders []string
+	for _, mibPath := range s.Path {
+		gosmi.AppendPath(mibPath)
+		folders = append(folders, mibPath)
+		filepath.Walk(mibPath, func(path string, info os.FileInfo, err error) error {
```

Comment:
I believe this folder walking code could be improved. I don't think it's truly recursive. For instance, if path is "/usr/share/snmp/mibs" and the user has mibs in "/usr/share/snmp/mibs/1/2/file.mib" it won't get picked up